src/App/Utilities/MapToolExporter.ts

```diff
@@ -1,113 +1,151 @@
-    if (index >= 0 && index < 8) {
-      xml += entry ({ key: attr[index], value: value.values.value.toString () })
-  }
-})
+function getStaticDataXML (hero: HeroModelRecord): string {
+  return "<id>"
+  + "<baGUID>eIj9Afv6STWmcKmhxyfaJQ==</baGUID>"
+  + "</id>"
+  + "<beingImpersonated>false</beingImpersonated>"
+  + "<exposedAreaGUID>"
+  + "<baGUID>CgAADK1LundsAAAAAgAADA==</baGUID>"
+  + "</exposedAreaGUID>"
+  + "<imageAssetMap>"
+  + "<entry>"
+  + "<null/>"
+  + "<net.rptools.lib.MD5Key>"
+  + "<id>f427aa49b28500217ac46ba37c2998a4</id>"
+  + "</net.rptools.lib.MD5Key>"
+  + "</entry>"
+  + "</imageAssetMap>"
+  + "<anchorX>0</anchorX>"
+  + "<anchorY>0</anchorY>"
+  + "<sizeScale>1.0</sizeScale>"
+  + "<snapToScale>true</snapToScale>"
+  + "<width>50</width>"
+  + "<height>50</height>"
+  + "<isoWidth>50</isoWidth>"
+  + "<isoHeight>50</isoHeight>"
+  + "<scaleX>1.0</scaleX>"
+  + "<scaleY>1.0</scaleY>"
+  + "<sizeMap>"
+  + "<entry>"
+  + "<java-class>net.rptools.maptool.model.SquareGrid</java-class>"
+  + "<net.rptools.maptool.model.GUID>"
+  + "<baGUID>CgAADK1LundsAAAAAgAADA==</baGUID>"
+  + "</net.rptools.maptool.model.GUID>"
+  + "</entry>"
+  + "</sizeMap>"
+  + "<snapToGrid>false</snapToGrid>"
+  + "<isVisible>true</isVisible>"
+  + "<visibleOnlyToOwner>false</visibleOnlyToOwner>"
+  + "<vblColorSensitivity>0</vblColorSensitivity>"
+  + "<alwaysVisibleTolerance>0</alwaysVisibleTolerance>"
+  + "<isAlwaysVisible>false</isAlwaysVisible>"
+  + `<name>${hero.values.name}</name>`
+  + "<ownerType>0</ownerType>"
+  + "<tokenShape>CIRCLE</tokenShape>"
+  + "<tokenType>PC</tokenType>"
+  + "<layer>TOKEN</layer>"
+  + "<propertyType>Basic</propertyType>"
+  + "<tokenOpacity>1.0</tokenOpacity>"
+  + "<terrainModifier>0.0</terrainModifier>"
+  + "<terrainModifiersIgnored>"
+  + "<net.rptools.maptool.model.Token_-TerrainModifierOperation>NONE"
+  + "</net.rptools.maptool.model.Token_-TerrainModifierOperation>"
+  + "</terrainModifiersIgnored>"
+  + "<isFlippedX>false</isFlippedX>"
+  + "<isFlippedY>false</isFlippedY>"
+  + "<hasSight>false</hasSight>"
+  + "<state/>"
+}
 
-  // TODO: folgende Werte irgendwie ausrechnen und entries machen
-  // const leP:number = Maybe.sum (fmapF (hero.race) (Race.A.lp))
-  // da hero.race ein string ist funktioniert das so nicht
-  const leP = 30, asp = 0, kap = 0
-  const schips = 3
-  const sk = 1, zk = 1
-  const ini = 12
-  const gs = 8
+function getAttributesXML (hero: HeroModelRecord, state: AppStateRecord): string {
+
+  // Hier werden die Attribute bestückt
+  // Irgendwas funktioniert aber noch nicht
+  let xml: string = pipe_ (state.values.wiki.values.attributes,
+    OrderedMap.elems,
+    fmap ((attr: Record<Attribute>) => entry ({
+      key: Attribute.A.short (attr),
+      value: pipe_ (attr,
+        Attribute.A.id,
+        OrderedMap.lookupF (hero.values.attributes)),
+    })),
+    entry,
+    List.intercalate (""))
```

Comment:
Vielleicht hilft dir das weiter …?

```suggestion
  let xml: string = pipe_ (
    state.values.wiki.values.attributes,
    OrderedMap.elems,
    mapMaybe ((attr: Record<Attribute>) => 
      pipe_ (
        attr,
        Attribute.A.id,
        OrderedMap.lookupF (hero.values.attributes),
        fmap (stateEntry => entry ({
          key: Attribute.A.short (attr),
          value: AttributeDependent.A.value (stateEntry),
        }))
      )
    ),
    List.intercalate (""))
```